include/libtorrent/aux_/suggest_piece.hpp

```diff
@@ -61,7 +61,7 @@ struct suggest_piece
 		// back and then reverse the items we put there.
 		for (int i = int(m_priority_pieces.size()) - 1; i >= 0; --i)
 		{
-			piece_index_t const piece = m_priority_pieces[i];
+			piece_index_t const piece = m_priority_pieces[std::size_t(i)];
```

Comment:
in the long run I would prefer to not have to do this. perhaps there could be another ``vector`` wrapper that accepts any integral type as index and hides the cast